debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/IncrementalSnapshotCollationSortOrderMismatchIT.java

```diff
@@ -0,0 +1,395 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.sqlserver;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import io.debezium.config.CommonConnectorConfig;
+import io.debezium.config.Configuration;
+import io.debezium.connector.sqlserver.util.TestHelper;
+import io.debezium.jdbc.JdbcConnection;
+import io.debezium.pipeline.source.snapshot.incremental.AbstractSnapshotTest;
+import io.debezium.util.Testing;
+
+/**
+ * Tests for <a href="https://issues.redhat.com/browse/DBZ-7359">DBZ-7359</a>
+ */
+public class IncrementalSnapshotCollationSortOrderMismatchIT extends AbstractSnapshotTest<SqlServerConnector> {
+    private static final int POLLING_INTERVAL = 1;
+    // Sort order of SQL_ collation does not correspond with Unicode
+    // see - https://learn.microsoft.com/en-us/sql/relational-databases/collations/collation-and-unicode-support?view=sql-server-ver16#SQL-collations
+    private static final String SQL_COLLATION = "SQL_Latin1_General_CP1_CI_AS";
+    // IDs that simulate the sort order mismatch between the SQL_ collation and Unicode.
+    private static final List<String> ALL_IDS = new ArrayList<>();
+    // Contains the IDs that get skipped when there is a sort order mismatch.
+    private static final List<String> SKIPPED_IDS = new ArrayList<>();
+
+    private SqlServerConnection connection;
+    private boolean isSendStringParametersAsUnicode;
+
+    @BeforeClass
+    public static void beforeClass() throws IOException {
+        try (InputStreamReader isr = new InputStreamReader(
```

Comment:
You can re-use `io.debezium.util.IoUtil` methods.

Reply:
done.